lib/models/os.ts

```diff
@@ -0,0 +1,674 @@
+	 * if it exists, or `null` is returned,
+	 * * `'latest'` in which case the most recent version is returned, including pre-releases,
+	 * * `'recommended'` in which case the recommended version is returned, i.e. the most
+	 * recent version excluding pre-releases, which can be `null` if only pre-release versions
+	 * are available,
+	 * * `'default'` in which case the recommended version is returned if available,
+	 * or `latest` is returned otherwise.
+	 * Defaults to `'latest'`.
+	 * @fulfil {String|null} - the version number, or `null` if no matching versions are found
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.os.getSupportedVersions('raspberry-pi').then(function(osVersions) {
+	 * 	console.log('Supported OS versions for raspberry-pi', osVersions);
+	 * });
+	 *
+	 * balena.models.os.getSupportedVersions('raspberry-pi', function(error, osVersions) {
+	 * 	if (error) throw error;
+	 * 	console.log('Supported OS versions for raspberry-pi', osVersions);
+	 * });
+	 */
+	const getMaxSatisfyingVersion = function(
+		deviceType: string,
+		versionOrRange: keyof OsVersions = 'latest',
+		callback?: (error?: Error, result?: string) => void,
+	): Promise<string> {
+		callback = findCallback(arguments);
+
+		return getValidatedDeviceType(deviceType)
+			.then(() => getSupportedVersions(deviceType))
+			.then(osVersions => _getMaxSatisfyingVersion(versionOrRange, osVersions))
+			.asCallback(callback);
+	};
+
+	/**
+	 * @summary Get the OS image last modified date
+	 * @name getLastModified
+	 * @public
+	 * @function
+	 * @memberof balena.models.os
+	 *
+	 * @param {String} deviceType - device type slug
+	 * @param {String} [version] - semver-compatible version or 'latest', defaults to 'latest'.
+	 * Unsupported (unpublished) version will result in rejection.
+	 * The version **must** be the exact version number.
+	 * To resolve the semver-compatible range use `balena.model.os.getMaxSatisfyingVersion`.
+	 * @fulfil {Date} - last modified date
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.os.getLastModified('raspberry-pi').then(function(date) {
+	 * 	console.log('The raspberry-pi image was last modified in ' + date);
+	 * });
+	 *
+	 * balena.models.os.getLastModified('raspberrypi3', '2.0.0').then(function(date) {
+	 * 	console.log('The raspberry-pi image was last modified in ' + date);
+	 * });
+	 *
+	 * balena.models.os.getLastModified('raspberry-pi', function(error, date) {
+	 * 	if (error) throw error;
+	 * 	console.log('The raspberry-pi image was last modified in ' + date);
+	 * });
+	 */
+	const getLastModified = function(
+		deviceType: string,
+		version: string = 'latest',
+		callback?: (error?: Error, result?: Date) => void,
+	): Promise<Date> {
+		callback = findCallback(arguments);
+
+		return getValidatedDeviceType(deviceType)
+			.then(() => normalizeVersion(version))
+			.then(ver =>
+				request.send({
+					method: 'HEAD',
+					url: deviceImageUrl(deviceType, ver),
+					baseUrl: apiUrl,
```

Comment:
Let's add back the comment that explains why this is not omitting the token, so that we don't forget about it.
> optionally authenticated, so we send the token in all cases